packages/react/src/components/dateInput/components/day/Day.tsx

```diff
@@ -54,9 +57,14 @@ export const Day = ({ day }: DayProps) => {
   const hasFocusedDate =
     focusedDate &&
     !isBefore(focusedDate, startOfMonth(currentMonth)) &&
-    !isAfter(focusedDate, endOfMonth(currentMonth));
+    !isAfter(focusedDate, endOfMonth(currentMonth)) &&
+    !isBlocked;
+
   if (
-    (!hasFocusedDate && (day.getDate() === 1 || day.getDate() === minDate.getDate())) ||
+    (!hasFocusedDate &&
+      (day.getDate() === 1 ||
+        day.getDate() === minDate.getDate() ||
+        day.getDate() === (currentMonthAvailableDates[0] && currentMonthAvailableDates[0].getDate()))) ||
```

Comment:
Small improvement suggestion: Since we only need the first available date of the month, maybe we could pass that to this component, instead of the whole list?